spacy/language.py

```diff
@@ -1150,17 +1150,12 @@ def update(
             component_cfg[name].setdefault("drop", drop)
             pipe_kwargs[name].setdefault("batch_size", self.batch_size)
         for name, proc in self.pipeline:
-            # ignore statements are used here because mypy ignores hasattr
-            if name not in exclude and hasattr(proc, "update"):
-                proc.update(examples, sgd=None, losses=losses, **component_cfg[name])  # type: ignore
-            if sgd not in (None, False):
-                if (
-                    name not in exclude
-                    and isinstance(proc, ty.TrainableComponent)
-                    and proc.is_trainable
-                    and proc.model not in (True, False, None)
-                ):
-                    proc.finish_update(sgd)
+            if (
+                name not in exclude
+                and isinstance(proc, ty.TrainableComponent)
+                and proc.is_trainable
```

Comment:
Under what circumstances is a `TrainableComponent`/`TrainablePipe` not trainable, i.e, `is_trainable` is `False`? There are no instances of pipes returning `False` in the codebase, as far as I can tell.

Reply:
Yeah, I was going back and forth on this as well. I was trying to think about cases where something is a `TrainableComponent` but is not trainable. The primary thing I could come up is a component that is used only during rehearsal or distillation, like our layer distiller. Such components could implement `TrainableComponent` by virtue of deriving from `TrainablePipe`, but are not trainable.